pydantic/_internal/_generate_schema.py

```diff
@@ -1891,20 +1891,21 @@ def _unsubstituted_typevar_schema(self, typevar: typing.TypeVar) -> core_schema.
 
         bound = typevar.__bound__
         constraints = typevar.__constraints__
+        typevar_default_value = getattr(typevar, '__default__', None)
 
         try:
             typevar_has_default = typevar.has_default()  # type: ignore
         except AttributeError:
             # could still have a default if it's an old version of typing_extensions.TypeVar
-            typevar_has_default = getattr(typevar, '__default__', None) is not None
+            typevar_has_default = typevar_default_value is not None
 
-        if (bound is not None) + (len(constraints) != 0) + typevar_has_default > 1:
+        if len(constraints) != 0 and (bound is not None or typevar_has_default):
             raise NotImplementedError(
-                'Pydantic does not support mixing more than one of TypeVar bounds, constraints and defaults'
+                'Pydantic does not support mixing more than one of TypeVar constraints and default/bound'
             )
 
         if typevar_has_default:
```

Comment:
I don't think we've fully added support here - if both `bound` and `default` are specified (and different), we need to figure out how to generate the right schema...